middleware.js

```diff
@@ -7,12 +7,12 @@ define(function(require) {
   return {
 
     requestTimeout : function(req, res, next) {
-
       var responded
+        , interval = req.timeout * 1000
```

Comment:
I know there’s some validation of this value in address.js, which is all well and good, but we should take care to validate these values here as well where they are actually used. If req.timeout isn’t a number, this will result in NaN and potentially difficult to debug behavior. Likewise, negative values can have confusing effects. We should take care here that the value must be a positive number, in the range `0-Infinity`, and if indeed it is not then and only then do we fall back to a standard timeout. That also implies that the standard value of 30sec shouldn’t be defined in address.js, but in this middleware and potentially exposed for the benefit of reuse by address.js.

Reply:
Also, we should add some tests to cover these inputs, particularly the cases of negative values an non-number values, including `null` and `undefined`.

Reply:
Added timeout validation and test cases.